TEE.md

```diff
@@ -0,0 +1,79 @@
+
+_Note: This key exchange protocol is on Curve25519, which is different from the Edwards curve used by NEAR._
+
+The key establishment scheme requires that the node as well as the back-up recipient have *mutually authenticated Curve25519 public keys*. The NEAR blockchain can be leveraged for this, more specifically:
+- the node operator whitelists a NEAR-account associated to the back-up recipient in the MPC smart contract.
+- the back-up recipient submits a public key (Curve25519) to the MPC smart contract. The smart contract only accepts it if the account has been whitelisted by the node operator.
+- the node submits a public key (Curve25519) to the MPC smart contract. The smart contract only accepts it if the public key submitting this transaction is an access key of the node operators account.
+
+_Note: Curve25519 and Edwards curve are [birationally equivalent](https://crypto.stackexchange.com/questions/43013/what-does-birational-equivalence-mean-in-a-cryptographic-context), so one could theoretically use NEAR account keys. But it seems cleaner if we just have the node and the back-up recipient generate dedicated keys for this._
+
+#### Encrypting the secret
+The node running inside the TEE monitors the contract and can derive an encryption key using its secret key and the public key submitted by the back-up recipient.
+
+#### Exporting the encrypted secret
+The back-up service triggers the export of the back-up through a call to the smart contract or a dedicated web-endpoint.
+The node returns the encrypted key either:
+- as a response to a request to its web-endpoint;
+- by printing its hex-value to the logs;
+- or through some other mechanism (@Barak, are there alternatives?)
+
+The back-up mechanism can then decrypt the received ciphertext and store the secret somewhere safe.
+
+_Todo: improve this section with more details. It is vague._
+
+### Restoration Mechanism:
+#### Encryption Key Agreement & Encrypting the secret 
+Same as above, just reversed roles
+
+#### Importing the encrypted secret into the node
+Similar to above, but reversed roles. The node makes a web request to the back-up service. (@Barak is this possible?)
+_Todo: improve this section with more details. It is vague._
+
+
+### Notes:
+- keep the web-endpoint private. Should we use a new webserver for that, on a different port?
+- instead of relying on web-endpoints, could we use TLS keys & connect via network, or connect via unix sockets?
+- for the hard-launch, the back-up service will need to run inside a TEE and must pass attestation verification on the contract.
```

Comment:
In the same contract I take it. I guess we'll need to elaborate more here, but we don't have to do this now as we're still lacking design docs for the original attestation flow (TBA in https://github.com/near/mpc/issues/857).

Reply:
For now, it probably has to be the same contract. In the future, we could add a different contract for that.
The main challenge for this would most likely be on the node. Idk how easy it is with the current code to monitor multiple contracts, but I would expect it to require quite a bit of effort.